src/match/serializers.py

```diff
@@ -0,0 +1,137 @@
+        if match.proposed_meeting_times is None:
+            return None
+        # We have to replace all single quotes with double quotes for JSON
+        proposed_meeting_times = json.loads(
+            match.proposed_meeting_times.replace("'", '"')
+        )
+        if proposed_meeting_times == []:
+            return None
+        return proposed_meeting_times
+
+    def get_proposed_locations(self, match):
+        proposed_locations = []
+        for location in match.proposed_locations.all():
+            proposed_locations.append(LocationSerializer(location).data)
+        if proposed_locations == []:
+            return None
+        return proposed_locations
+
+    def get_meeting_location(self, match):
+        meeting_location = LocationSerializer(match.meeting_location).data
+        if meeting_location == {}:
+            return None
+        return meeting_location
+
+    class Meta:
+        model = Match
+        fields = (
+            "id",
+            "status",
+            "matched_user",
+            "proposer_id",
+            "accepted_ids",
+            "proposed_meeting_times",
+            "proposed_locations",
+            "meeting_location",
+            "meeting_time",
+        )
+        read_only_fields = fields
+
+
+class BothUsersMatchSerializer(serializers.ModelSerializer):
+    users = serializers.SerializerMethodField("get_users")
+    accepted_ids = serializers.SerializerMethodField("get_accepted_ids")
+    proposed_meeting_times = serializers.SerializerMethodField(
+        "get_proposed_meeting_times"
+    )
+    proposed_locations = serializers.SerializerMethodField("get_proposed_locations")
+    meeting_location = serializers.SerializerMethodField("get_meeting_location")
+
+    def get_users(self, match):
+        return (
+            SimpleUserSerializer(match.user_1).data,
+            SimpleUserSerializer(match.user_2).data,
+        )
+
+    def get_accepted_ids(self, match):
+        if match.accepted_ids is None:
+            return None
+        accepted_ids = json.loads(match.accepted_ids)
+        if accepted_ids == []:
+            return None
+        return accepted_ids
+
+    def get_proposed_meeting_times(self, match):
+        if match.proposed_meeting_times is None:
+            return None
+        proposed_meeting_times = json.loads(
+            match.proposed_meeting_times.replace("'", '"')
+        )
+        if proposed_meeting_times == []:
+            return None
+        return proposed_meeting_times
+
+    def get_proposed_locations(self, match):
+        proposed_locations = []
+        for location in match.proposed_locations.all():
+            proposed_locations.append(LocationSerializer(location).data)
+        if proposed_locations == []:
+            return None
+        return proposed_locations
+
+    def get_meeting_location(self, match):
+        meeting_location = LocationSerializer(match.meeting_location).data
+        if meeting_location == {}:
```

Comment:
wait why does this matter

Reply:
cuz if meeting location is none in django it's an empty python set but i still wanna just return `None`, but I'm gonna do ur cool `return meeting_location or None`

Reply:
haha but don't take my word for it, i might be wrong :p 

yah im just being picky